parser.py

```diff
@@ -22,52 +30,28 @@ def headlines(design = "APEX"):
     try:
         mass = 0
         for element in rocket.iter('overridemass'):
-            mass = mass + float(element.text)
+            mass += float(element.text)
         for element in rocket.iter('mass'):
-            mass = mass + float(element.text)
+            mass += float(element.text)
         mass = round(mass, 3) # we can only measure to nearest gram anyway
     except:
         mass = "Mass calculation not available"
 
     # extract simulation data
-    try:
-        apogee = sims[0].find('flightdata').attrib['maxaltitude']        
-    except:
-        apogee = "Error parsing simulation data"
-
-    try:
-        maxvel = sims[0].find('flightdata').attrib['maxvelocity']
-    except:
-        maxvel = "Error parsing simulation data"
-
-    try:
-        maxacc = sims[0].find('flightdata').attrib['maxacceleration']
-    except:
-        maxacc = "Error parsing simulation data"
-
-    try:
-        flightdur = sims[0].find('flightdata').attrib['flighttime']
-    except:
-        flightdur = "Error parsing simulation data"
-
-    try:
-        groundhit = sims[0].find('flightdata').attrib['groundhitvelocity']
-    except:
-        groundhit = "Error parsing simulation data"
+    apogee    = simdata (sims,'flightdata','maxaltitude')
+    maxvel    = simdata (sims, 'flightdata','maxvelocity')
+    maxacc    = simdata (sims, 'flightdata','maxacceleration')
+    flightdur = simdata (sims, 'flightdata', 'flighttime') 
+    groundhit = simdata (sims, 'flightdata', 'groundhitvelocity')
+    maxmach   = simdata (sims,'flightdata', 'maxmach')
+    timetoapp = simdata (sims, 'flightdata', 'timetoapogee')
 
     # other extractable params - max mach, time to apogee, launch rod vel, deployment vel
     # see raw xml to get attrib keys
 
     # pretty print data
-    summary = str()
-
-    summary += f"**Motor selected for use:** {motor} <br/> \n"
-    summary += f"**Apogee:** {apogee} m <br/> \n"
-    summary += f"**Max speed:** {maxvel} m/s <br/> \n"
-    summary += f"**Max acceleration:** {maxacc} m/s^2 <br/> \n"
-    summary += f"**Flight duration:** {flightdur} s <br/> \n"
-    summary += f"**Ground hit velocity:** {groundhit} m/s <br/> \n"
-    summary += f"**Dry mass:** {mass} kg "
+    sumlist = [f"**Motor selected for use:** {motor} <br/> \n", f"**Apogee:** {apogee} m <br/> \n", f"**Max speed:** {maxvel} m/s <br/> \n", f"**Max acceleration:** {maxacc} m/s^2 <br/> \n", f"**Flight duration:** {flightdur} s <br/> \n", f"**Ground hit velocity:** {groundhit} m/s <br/> \n", f"**Dry mass:** {mass} kg "]
+    summary = sumlist.join('')
```

Comment:
I like the functional programming, but the other one was cleaner rip

Reply:
can we keep the legible version lol i want to be able to edit this without having to use duolingo

Reply:
yeah we should keep it clean